e2e_test.go

```diff
@@ -840,3 +841,39 @@ func TestE2EServerTimeouts(t *testing.T) {
 		}
 	})
 }
+
+// TestE2EMaxConnectionAge_ConnectionAgeSet tests that when using the
+// WithMaxConnectionAge server option the connection age is available
+// to the handler. This is a fairly weak test as it doesn't actually
+// test that connections are closed. Unfortunately our test framework
+// seems to always close connections so this is as good as we can get.
+func TestE2EMaxConnectionAge_ConnectionAgeSet(t *testing.T) {
+	addConnectionStartTimeHeader = true
+
+	someFlavours(t, []string{
+		"http1.1",
+		"http1.1-tls",
+		"http2.0-h2",
+
+		// The Go h2c implementation doesn't currently support max
+		// connection age.
```

Comment:
Same as the other server options – not supported by h2c (which we don't use anywhere).